### PR TITLE
make browsershot wait until network idle on ExecuteJavascriptMiddleware

### PR DESCRIPTION
**Pull request by chrismcintosh**
Hi Kai,

As requested here is the pull request for making Browsershot wait until network idle on the ExecuteJavascriptMiddleware.

This is my first Open Source pull request - so thanks for your patience!

Best,
Chris

### CODE REVIEW

**Comment by ksassnowski** [comment]
Psalm seems to be complaining about this change. I had a quick look at it and it looks like we just have to suppress the error. The problem is that the return type of `waitUntilNetworkIdle` method of the `Browsershot` class isn't properly annotated, so Psalm can't guarantee that the closure actually returns the correct type. Since this is not something we can fix in this repository, I'm fine with ignoring the error for now.

File: src/Downloader/Middleware/ExecuteJavascriptMiddleware.php [hunk 1]
```diff
@@ -35,7 +35,7 @@ public function __construct(
         private LoggerInterface $logger,
         ?callable $getBrowsershot = null,
     ) {
-        $this->getBrowsershot = $getBrowsershot ?: static fn (string $uri): Browsershot => Browsershot::url($uri);
+        $this->getBrowsershot = $getBrowsershot ?: static fn (string $uri): Browsershot => Browsershot::url($uri)->waitUntilNetworkIdle();
```
Comment (ksassnowski) [inline src/Downloader/Middleware/ExecuteJavascriptMiddleware.php:38]:
```
/** @psalm-suppress MixedInferredReturnType, MixedReturnStatement */
```

Add this above line 38 to suppress the Psalm warning. You can check that the error went away locally by running `composer analyze`.

Comment (chrismcintosh) [reply src/Downloader/Middleware/ExecuteJavascriptMiddleware.php:39]: Confirmed. Thanks!

**Comment by ksassnowski** [comment]
Fixes #56 

Alright, looks good to me! Thanks a lot for the contribution and congrats on your first open source contribution! I'll tag a new release with change later 👍

**Pull request merged**